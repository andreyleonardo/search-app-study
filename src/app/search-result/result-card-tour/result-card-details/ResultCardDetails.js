import React from 'react';
import PropTypes from 'prop-types';
import Ratings from '../../../ratings/Ratings';

import './ResultCardDetails.scss';

const ResultCardDetails = ({
  rating, isSpecialOffer, currency, price
}) => (
  <section className="result_card_details row between-xs middle-xs">
    <div className="result_card_details__rating col-xs-4 col-sm">
      <Ratings rating={rating}>
        <span>{`${rating}/5`}</span>
      </Ratings>
    </div>
    <div className="result_card_details__pricing col-xs col-sm">
      {isSpecialOffer && <span className="result_card_details__special_offer">Special Offer</span>}
      <span className="result_card_details__price">{currency + price}</span>
    </div>
  </section>
);

ResultCardDetails.propTypes = {
  rating: PropTypes.number,
  currency: PropTypes.string.isRequired,
  price: PropTypes.string.isRequired,
  isSpecialOffer: PropTypes.bool
};

ResultCardDetails.defaultProps = {
  rating: 0,
  isSpecialOffer: false
};

export default ResultCardDetails;
