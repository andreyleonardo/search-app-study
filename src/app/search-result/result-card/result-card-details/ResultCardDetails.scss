@import "src/stylesheets/css/_colors";

.result_card_details {
  &__rating, &__pricing {
    margin-top: 20px;
  }

  &__pricing {
    text-align: right;
  }

  &__special_offer {
    background-color: $orange;
    color: white;
    font-size: 16px;
    border-radius: 3px;
    justify-content: center;
    align-items: center;
    padding: 5px;
    vertical-align: middle;
  }

  &__price {
    right: 10px;
    vertical-align: middle;
    margin: 3px;
  }
}
