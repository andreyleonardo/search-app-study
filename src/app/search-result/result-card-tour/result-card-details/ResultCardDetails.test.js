import React from 'react';
import ResultCardDetails from './ResultCardDetails';

describe('Result Card Details', () => {
  let props;
  let resultCardDetails;

  beforeEach(() => {
    props = {
      currency: '$',
      price: '30',
      rating: 4.8
    };
    resultCardDetails = shallow(<ResultCardDetails {...props} />);
  });

  it('renders without crash', () => {
    expect(resultCardDetails).toMatchSnapshot();
  });

  it('renders Special Offer text when isSpecialOffer is true', () => {
    resultCardDetails.setProps({ isSpecialOffer: true });

    expect(resultCardDetails).toMatchSnapshot();
  });
});
