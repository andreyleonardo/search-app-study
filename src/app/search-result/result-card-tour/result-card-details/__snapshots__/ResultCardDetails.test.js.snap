// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Result Card Details renders Special Offer text when isSpecialOffer is true 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ResultCardDetails
    currency="$"
    isSpecialOffer={true}
    price="30"
    rating={4.8}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="result_card_details__rating col-xs-4 col-sm"
        >
          <Ratings
            rating={4.8}
          >
            <span>
              4.8/5
            </span>
          </Ratings>
        </div>,
        <div
          className="result_card_details__pricing col-xs col-sm"
        >
          <span
            className="result_card_details__special_offer"
          >
            Special Offer
          </span>
          <span
            className="result_card_details__price"
          >
            $30
          </span>
        </div>,
      ],
      "className": "result_card_details row between-xs middle-xs",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Ratings
            rating={4.8}
          >
            <span>
              4.8/5
            </span>
          </Ratings>,
          "className": "result_card_details__rating col-xs-4 col-sm",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <span>
              4.8/5
            </span>,
            "rating": 4.8,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "4.8/5",
            },
            "ref": null,
            "rendered": "4.8/5",
            "type": "span",
          },
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span
              className="result_card_details__special_offer"
            >
              Special Offer
            </span>,
            <span
              className="result_card_details__price"
            >
              $30
            </span>,
          ],
          "className": "result_card_details__pricing col-xs col-sm",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Special Offer",
              "className": "result_card_details__special_offer",
            },
            "ref": null,
            "rendered": "Special Offer",
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "$30",
              "className": "result_card_details__price",
            },
            "ref": null,
            "rendered": "$30",
            "type": "span",
          },
        ],
        "type": "div",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="result_card_details__rating col-xs-4 col-sm"
          >
            <Ratings
              rating={4.8}
            >
              <span>
                4.8/5
              </span>
            </Ratings>
          </div>,
          <div
            className="result_card_details__pricing col-xs col-sm"
          >
            <span
              className="result_card_details__special_offer"
            >
              Special Offer
            </span>
            <span
              className="result_card_details__price"
            >
              $30
            </span>
          </div>,
        ],
        "className": "result_card_details row between-xs middle-xs",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Ratings
              rating={4.8}
            >
              <span>
                4.8/5
              </span>
            </Ratings>,
            "className": "result_card_details__rating col-xs-4 col-sm",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <span>
                4.8/5
              </span>,
              "rating": 4.8,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "4.8/5",
              },
              "ref": null,
              "rendered": "4.8/5",
              "type": "span",
            },
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                className="result_card_details__special_offer"
              >
                Special Offer
              </span>,
              <span
                className="result_card_details__price"
              >
                $30
              </span>,
            ],
            "className": "result_card_details__pricing col-xs col-sm",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Special Offer",
                "className": "result_card_details__special_offer",
              },
              "ref": null,
              "rendered": "Special Offer",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "$30",
                "className": "result_card_details__price",
              },
              "ref": null,
              "rendered": "$30",
              "type": "span",
            },
          ],
          "type": "div",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Result Card Details renders without crash 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ResultCardDetails
    currency="$"
    isSpecialOffer={false}
    price="30"
    rating={4.8}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="result_card_details__rating col-xs-4 col-sm"
        >
          <Ratings
            rating={4.8}
          >
            <span>
              4.8/5
            </span>
          </Ratings>
        </div>,
        <div
          className="result_card_details__pricing col-xs col-sm"
        >
          <span
            className="result_card_details__price"
          >
            $30
          </span>
        </div>,
      ],
      "className": "result_card_details row between-xs middle-xs",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Ratings
            rating={4.8}
          >
            <span>
              4.8/5
            </span>
          </Ratings>,
          "className": "result_card_details__rating col-xs-4 col-sm",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <span>
              4.8/5
            </span>,
            "rating": 4.8,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "4.8/5",
            },
            "ref": null,
            "rendered": "4.8/5",
            "type": "span",
          },
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            false,
            <span
              className="result_card_details__price"
            >
              $30
            </span>,
          ],
          "className": "result_card_details__pricing col-xs col-sm",
        },
        "ref": null,
        "rendered": Array [
          false,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "$30",
              "className": "result_card_details__price",
            },
            "ref": null,
            "rendered": "$30",
            "type": "span",
          },
        ],
        "type": "div",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="result_card_details__rating col-xs-4 col-sm"
          >
            <Ratings
              rating={4.8}
            >
              <span>
                4.8/5
              </span>
            </Ratings>
          </div>,
          <div
            className="result_card_details__pricing col-xs col-sm"
          >
            <span
              className="result_card_details__price"
            >
              $30
            </span>
          </div>,
        ],
        "className": "result_card_details row between-xs middle-xs",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Ratings
              rating={4.8}
            >
              <span>
                4.8/5
              </span>
            </Ratings>,
            "className": "result_card_details__rating col-xs-4 col-sm",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <span>
                4.8/5
              </span>,
              "rating": 4.8,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "4.8/5",
              },
              "ref": null,
              "rendered": "4.8/5",
              "type": "span",
            },
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              false,
              <span
                className="result_card_details__price"
              >
                $30
              </span>,
            ],
            "className": "result_card_details__pricing col-xs col-sm",
          },
          "ref": null,
          "rendered": Array [
            false,
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "$30",
                "className": "result_card_details__price",
              },
              "ref": null,
              "rendered": "$30",
              "type": "span",
            },
          ],
          "type": "div",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
